### ISSUE TITLE
Errors with Sublime Text 3 and TypeScript package

### ISSUE THREAD

**Issue by mc100s** [mgechev/angular2-seed#1404]
Hello,

I try to start a new project in Angular 2 on Windows based on this repository. Everything works fine (`npm install`, `npm start`, ...) except that Sublime Text 3 (with the [_TypeScript_ package][1]) shows me some errors that are not present when I execute the project.

Here are the steps I followed:

    $ git clone https://github.com/mgechev/angular2-seed.git
    $ cd angular2-seed
    $ npm install

# Error 1
 
Classes with `@Component` decorators show the following error:

> Experimental support for decorators is a feature that is subject to change in a future release. Set the 'experimentalDecorators' option to remove this warning.

# Error 2

In _src/client/app/about/about.component.ts_ for example, on the line:

      moduleId: module.id,

Sublime shows this error:

> Cannot find name 'module'.

# The _tsconfig.json_

    {
      "compilerOptions": {
        "target": "es5",
        "module": "commonjs",
        "declaration": false,
        "removeComments": true,
        "noLib": false,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "lib": ["es6", "es2015", "dom"],
        "sourceMap": true,
        "pretty": true,
        "allowUnreachableCode": false,
        "allowUnusedLabels": false,
        "noImplicitAny": true,
        "noImplicitReturns": true,
        "noImplicitUseStrict": false,
        "noFallthroughCasesInSwitch": true,
        "typeRoots": [
          "./node_modules/@types",
          "./node_modules"
        ],
        "types": [
          "node"
        ]
      },
      "exclude": [
        "node_modules",
        "dist",
        "src"
      ],
      "compileOnSave": false
    }

---------

I think Sublime Text ignores the line `"experimentalDecorators": true` in _tsconfig.json_, even if the TypeScript package says it supports it: https://github.com/Microsoft/TypeScript-Sublime-Plugin#configured-project

Do you why I have this 2 errors? I can develop but it's not very pleasant.

Thank you in advance!

  [1]: https://github.com/Microsoft/TypeScript-Sublime-Plugin

**Comment by TheDonDope** [comment]
Hi @mc100s 
I am kind of in the similar environment here.
I am developing on MacOS Sierra with Sublime Text 3 (currently: Dev Channel, Build 3125) and the Sublime TypeScript Package (v0.1.16).

A bit of history:

Originally, there was only one `tsconfig.json`, the one in the root directory.
Then, there was issue #719 where Visual Studio Code required a separate `tsconfig.json` in `/src` (see: https://github.com/mgechev/angular2-seed/commit/ed4a763e2a6ddf792eeee9695d5f7a52eae19880)

Long story short: I am facing the same issue still :/

**Comment by mgechev** [comment]
If I finish on time with codelyzer early enough this weekend, I'll take this issue.

**Comment by TheDonDope** [comment]
Weird, i just tried on the latest HEAD (b7b43ed7b34b8112c72a06b00f209baf7593549e) and cannot reproduce it anymore. The errors do not show up anymore. 😕

**Comment by mgechev** [comment]
Sounds awesome (and scary) :-). Did you update sublime?

**Comment by mc100s** [comment]
I use the Stable Channel, Build 3126 of Sublime Text.
And I also use TypeScript Package v0.1.16.

I tried with the latest build but still the same issue :(

**Comment by mc100s** [comment]
I solved my issue thanks to this link: https://github.com/Microsoft/TypeScript-Sublime-Plugin/issues/470

To summarize:  
- Go to the User preferences (Ctrl + Shift + P > _Preferences: Settings_)
- Add the following line: `"typescript_tsdk": "C:/...../npm/node_modules/typescript/lib",` (with the appropriate link)
- Restart Sublime Text
- Test if it works in  _about.component.ts_ file by pressing F12 when the cursor is on `module.id`
- Enjoy!

**Comment by mgechev** [comment]
@mc100s thanks for sharing the solution!
